src/hal/src/window.rs

```diff
@@ -414,7 +444,7 @@ pub trait Swapchain<B: Backend>: Any + Send + Sync {
         timeout_ns: u64,
         semaphore: Option<&B::Semaphore>,
         fence: Option<&B::Fence>,
-    ) -> Result<SwapImageIndex, AcquireError>;
+    ) -> Result<(SwapImageIndex, Option<Suboptimal>), AcquireError>;
```

Comment:
would you consider instead returning a struct like
```rust
struct AcquireFrame {
  index: SwapImageIndex,
  suboptimal: bool,
}
```

Reply:
And for `present`?

Reply:
Oh, this is a bit unfortunate. Would it be simpler if we just add `is_optimal` method to the `Swapchain`? It's not an error here, so we aren't forcing anyone to check for it anyway.

Reply:
It is not an error, but it **should** be checked, and making it part of return type kinda encourages user to do the check.